data/moves.ts

```diff
@@ -7305,6 +7450,35 @@ export const BattleMovedex: {[moveid: string]: MoveData} = {
 		zMove: {boost: {spe: 1}},
 		contestType: "Clever",
 	},
+	grassyglide: {
+		// NOT CONFIRMED
+		num: 99999,
+		// NOT CONFIRMED
+		accuracy: 100,
+		// NOT CONFIRMED
+		basePower: 80,
+		// NOT CONFIRMED
+		category: "Physical",
+		desc: "If this move is used while Grassy Terrain is active, its user will nearly always move first.",
+		shortDesc: "+2 Priority under Grassy Terrain.",
+		name: "Grassy Glide",
+		// NOT CONFIRMED
+		pp: 5,
+		priority: 0,
+		// NOT CONFIRMED
+		flags: {contact: 1, protect: 1},
+		onModifyMove(move, pokemon) {
+			if (this.field.getTerrain().name === "Grassy Terrain") {
+				move.priority = 2;
+			}
+		},
```

Comment:
Priority is calculated in `getActionSpeed` which happens before `onModifyMove`.

(I assume here that you can switch in a Grassy Surge user and get priority on your ally's Grassy Glide.)

Reply:
We don’t know until we can test. I’ll make a change when we have more info.

Reply:
Your latest code is now what I was originally expecting, so that's all good.